dll/win32/shell32/shlexec.cpp

```diff
@@ -1738,6 +1738,59 @@ static BOOL SHELL_translate_idlist(LPSHELLEXECUTEINFOW sei, LPWSTR wszParameters
     return appKnownSingular;
 }
 
+static BOOL
+SHELL_OpenPIDL(
+    _In_ LPSHELLEXECUTEINFOW sei,
+    _In_ LPCITEMIDLIST pidl)
+{
+    // Bind pidl
+    CComPtr<IShellFolder> psfFolder;
+    LPCITEMIDLIST pidlLast;
+    HRESULT hr = SHBindToParent(pidl, IID_PPV_ARG(IShellFolder, &psfFolder), &pidlLast);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return FALSE;
+
+    // Get the context menu to invoke a command
+    CComPtr<IContextMenu> pCM;
+    hr = psfFolder->GetUIObjectOf(NULL, 1, &pidlLast, IID_NULL_PPV_ARG(IContextMenu, &pCM));
+    if (FAILED_UNEXPECTEDLY(hr))
+        return FALSE;
+
+    // Invoke a command
+    CMINVOKECOMMANDINFO ici = { sizeof(ici) };
+    ici.fMask = (sei->fMask & (SEE_MASK_NO_CONSOLE | SEE_MASK_ASYNCOK | SEE_MASK_FLAG_NO_UI));
+    ici.nShow = sei->nShow;
+    ici.hwnd = sei->hwnd;
+    char szVerb[64];
+    if (sei->lpVerb)
+    {
```

Comment:
Also verify that the verb is not empty by checking `[0]`?

Reply:
Fixed.